src/Tizen.Sensor/Tizen.Sensor/SensorRecorder.cs

```diff
@@ -0,0 +1,504 @@
+        /// <param name="key">
+        /// Data attribute to retrieve.
+        /// </param>
+        /// <returns>
+        /// returns a double value from a record data.
+        /// </returns>
+        /// <since_tizen> 6 </since_tizen>
+        public double DataGetDouble(int data, int key)
+        {
+            double value;
+            int error = Interop.SensorRecoder.DataGetDouble(data, key, out value);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error getting the sensor recorder double data");
+                throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorder.DataGetDouble Failed");
+            }
+            return value;
+        }
+
+        /// <summary>
+        /// Gets the start and the end time of the time period of a given record data.
+        /// </summary>
+        /// <param name="data">
+        /// Record data handle.
+        /// </param>
+        /// <returns>
+        /// returns the start and the end time of the time period of a given record data.
+        /// </returns>
+        /// <since_tizen> 6 </since_tizen>
+        public Tuple<long, long> DataGetTime(int data)
+        {
+            long startTime, endTime;
+
+            int error = Interop.SensorRecoder.DataGetTime(data, out startTime, out endTime);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error getting the sensor recorder time");
+                throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorder.DataGetTime Failed");
+            }
+
+            Tuple<long, long> list = new Tuple<long, long>(startTime, endTime);
+            return list;
+        }
+
+        /// <summary>
+        /// Checks whether it is supported to record a given sensor type.
+        /// </summary>
+        /// <param name="type">
+        /// A sensor type to check.
+        /// </param>
+        /// <returns>
+        /// returns true if it is supported to record a given sensor type.
+        /// </returns>
+        /// <since_tizen> 6 </since_tizen>
+        public bool IsSupported(int type)
+        {
+            bool isSupported;
+            int error = Interop.SensorRecoder.IsSupported(type, out isSupported);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error checking if sensor recorder is supported");
+                throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorder.IsSupported Failed");
+            }
+            return isSupported;
+        }
+
+        /// <summary>
+        /// Sets a recording option parameter.
+        /// </summary>
+        /// <param name="recorderOption">
+        /// Option parameter.
+        /// </param>
+        /// <param name="value">
+        /// Value.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void OptionSetInt(int recorderOption, int value)
```

Comment:
Do not use int for option
I can't know which options are usable.

This API is too low level to use

Reply:
There are 2 options 
SENSOR_RECORDER_OPTION_RETENTION_PERIOD
SENSOR_RECORDER_OPTION_INTERVAL
should I replace int with enum ?

Reply:
Sure